embedded/sql/stmt.go

```diff
@@ -1648,6 +1718,116 @@ func (v *SysFn) selectorRanges(table *Table, asTable string, params map[string]i
 	return nil
 }
 
+type Cast struct {
+	val ValueExp
+	t   SQLValueType
+}
+
+func (c *Cast) getConverter(src, dst SQLValueType) func(TypedValue) (TypedValue, error) {
+	if dst == TimestampType {
+
+		if src == IntegerType {
+			return func(val TypedValue) (TypedValue, error) {
+				return &Timestamp{val: time.Unix(0, val.Value().(int64)).UTC()}, nil
+			}
+		}
+
+		if src == VarcharType {
+			return func(val TypedValue) (TypedValue, error) {
+				str := val.Value().(string)
+				for _, layout := range []string{
+					"2006-01-02 15:04:05.999999999",
+					"2006-01-02 15:04",
+					"2006-01-02",
+				} {
+					t, err := time.ParseInLocation(layout, str, time.UTC)
+					if err == nil {
+						return &Timestamp{val: t.UTC()}, nil
+					}
+				}
+
+				return nil, fmt.Errorf(
+					"%w: can not cast string '%s' as a TIMESTAMP",
+					ErrIllegalArguments,
+					str[:100],
```

Comment:
100?

Reply:
To avoid too long strings ending up in the error message (should add some ... at the end but was to lazy ;))

Reply:
Changed the code a bit to be easier to understand